src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -859,5 +846,39 @@ protected override void Dispose(bool disposing)
             var props = await CommonProjectServices.ActiveConfiguredProjectProperties.GetProjectDebuggerPropertiesAsync().ConfigureAwait(false);
             await props.ActiveDebugProfile.SetValueAsync(profileName).ConfigureAwait(true);
         }
+
+        internal async Task<string> GetLaunchSettingsFileAsync()
+        {
+            if (_launchSettingsFile == null)
```

Comment:
Do we potentially have lots of threads calling this at the same time?  You can consider using AsyncLazy<> in the VS threading library, this will ensure the calculation done once, and when multiple threads calling it at it, they will wait the same task to finish.